### PR TITLE
OCPBUGS-20016: improve labels and annotations modal display and editi…

### PR DESCRIPTION
**Pull request by rhamilto**
…ng behavior

Videos of the integration test additions demonstrating new behavior:
* if a label or annotation is added while the modal is open the Save button is disabled and an alert displays 

https://github.com/openshift/console/assets/895728/aaea1b7d-f4b9-4448-abca-afbdae9dc4d6

https://github.com/openshift/console/assets/895728/7ad6569d-7af9-4c0f-84a8-cf0b982f9ae8

### CODE REVIEW

**Review by TheRealJon — CHANGES_REQUESTED** [review]
I was poking around at this and found what might be a bug:

1. Open the modal
2. Make a background change (which propagates to the modal)
3. Edit a label in the modal

Result: the  "stale" behavior is triggered as soon as an edit is made in the modal.

Solving this will probably be overly complex, so we can stick to your original plan and remove the logic that propagates changes into the modal. Let me know what you think.

**Comment by rhamilto** [comment]
> Result: the "stale" behavior is triggered as soon as an edit is made in the modal.

Great catch.  Let me simplify.

**Comment by rhamilto** [comment]
Updated, @TheRealJon.  PTAL.

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Comment by rhamilto** [comment]
/hold

for a broken knative integration test

**Comment by rhamilto** [comment]
/hold cancel

Thank you, integration test, for uncovering my bugs.

**Comment by rhamilto** [comment]
@TheRealJon, mind retagging?

File: frontend/public/components/modals/labels-modal.tsx [hunk 1]
```diff
@@ -35,9 +36,21 @@ const BaseLabelsModal: React.FC<BaseLabelsModalProps> = ({
   const [labels, setLabels] = React.useState(
     SelectorInput.arrayify(_.get(resource, path.split('/').slice(1))),
   );
+  const [watchedResource, watchedResourceLoaded] = useK8sWatchResource<K8sResourceCommon>({
+    kind: resource.kind,
+    name: resource?.metadata?.name,
+    namespace: resource?.metadata?.namespace,
+  });
```
Comment (TheRealJon) [inline frontend/public/components/modals/labels-modal.tsx:43]: If we use optional chaining for the other properties, we probably need it here too:
```suggestion
  const [watchedResource, watchedResourceLoaded] = useK8sWatchResource<K8sResourceCommon>({
    kind: resource?.kind,
    name: resource?.metadata?.name,
    namespace: resource?.metadata?.namespace,
  });
```

File: frontend/public/components/modals/tags.tsx [hunk 2]
```diff
@@ -34,9 +35,20 @@ export const TagsModal = withHandlePromise<TagsModalProps & HandlePromiseProps>(
     _.isEmpty(props.tags) ? [['', '']] : _.toPairs(props.tags),
   );
   const [errorMessage, setErrorMessage] = React.useState(props.errorMessage);
-
+  const [watchedResource, watchedResourceLoaded] = useK8sWatchResource<K8sResourceCommon>({
+    kind: props.resource.kind,
```
Comment (TheRealJon) [inline frontend/public/components/modals/tags.tsx:39]:
```suggestion
    kind: props.resource?.kind,
```

**Review by TheRealJon — CHANGES_REQUESTED** [review]
Just a couple of small adjustments!

File: frontend/public/components/modals/labels-modal.tsx (hunk 1 above)
Comment (rhamilto) [reply frontend/public/components/modals/labels-modal.tsx:43]: We have instances all over the code base there `kind` and `metadata` don't use optional chaining.  But now the linter complains about `metadata` but not `kind`.  Wonder what changed?  It wasn't https://github.com/openshift/console/blame/af5b08f9be05e01333b4595c11006657fe952ceb/frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts#L60-L61

File: frontend/public/components/modals/tags.tsx (hunk 2 above)
Comment (rhamilto) [reply frontend/public/components/modals/tags.tsx]: Updated

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Pull request merged**